processing/src/main/java/org/apache/carbondata/processing/loading/steps/WriterProcessorStepImpl.java

```diff
@@ -271,37 +262,18 @@ private CarbonRow convertRow(CarbonRow row) throws KeyGenException {
   }
 
   private void processBatch(CarbonRowBatch batch, CarbonFactHandler dataHandler, int iteratorIndex)
-      throws CarbonDataLoadingException {
-    try {
-      while (batch.hasNext()) {
-        CarbonRow row = batch.next();
-        CarbonRow converted = convertRow(row);
-        dataHandler.addDataToStore(converted);
-        readCounter[iteratorIndex]++;
+  {
+    while (batch.hasNext()) {
+      CarbonRow row;
+      if (noSort) {
```

Comment:
move 'if' statement to the outside of 'while' loop

Reply:
ok, fixed